### PR TITLE
[FLINK-8116] Stale comments referring to Checkpointed interface

### PR DESCRIPTION
**Pull request by ankitiitb1069**
## What is the purpose of the change

This pull request replaces the stale comments referring to Checkpointed interface with the new one defined as CheckpointedFunction interface

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no 

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java [hunk 1]
```diff
@@ -34,17 +34,17 @@
...
+ * <p>Sources that also implement the {@link org.apache.flink.streaming.api.checkpoint.CheckpointedFunction}
  * interface must ensure that state checkpointing, updating of internal state and emission of
  * elements are not done concurrently. This is achieved by using the provided checkpointing lock
  * object to protect update of state and emission of elements in a synchronized block.
  *
  * <p>This is the basic pattern one should follow when implementing a (checkpointed) source:
  *
  * <pre>{@code
- *  public class ExampleSource<T> implements SourceFunction<T>, Checkpointed<Long> {
+ *  public class ExampleSource<T> implements SourceFunction<T>, CheckpointedFunction<Long> {
```
Comment (ggevay) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java]: Could you please also update the methods of the class in the code examples? `snapshotState` and `restoreState` have changed with the interface change.

Comment (ankitiitb1069) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java]: Please check, I have made the changes, but could not make up what to write inside of these functions

Comment (ggevay) [reply flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java]: You can see the function bodies here:
https://ci.apache.org/projects/flink/flink-docs-release-1.4/dev/stream/state/state.html#stateful-source-functions

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java [hunk 2]
```diff
@@ -61,9 +61,9 @@
  *          isRunning = false;
  *      }
  *
- *      public Long snapshotState(long checkpointId, long checkpointTimestamp) { return count; }
+ *      public void snapshotState(FunctionSnapshotContext context) {  }
```
Comment (tzulitai) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java]: These methods should handle checkpointing of the `count` state.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java [hunk 3]
```diff
@@ -61,9 +61,9 @@
...
  *
- *      public void restoreState(Long state) { this.count = state; }
+ *      public void initializeState(FunctionInitializationContext context) {  }
```
Comment (tzulitai) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java]: These methods should handle checkpointing of the count state.

**Comment by tzulitai** [comment]
Thanks @ankitiitb1069 @ggevay for the work and review.
The changes LGTM, minus my comment. I'll address my comments while merging this ...

**Pull request closed without merging**